extras/dagger/main.go

```diff
@@ -0,0 +1,277 @@
+			"--name", name,
+			"--value", mountPath,
+		}).
+		Sync(ctx)
+	return att, err
+}
+
+func (att *Attestation) Debug() *Terminal {
+	return att.
+		Container(0).
+		Terminal()
+}
+
+// Build an ephemeral container with everything needed to process the attestation
+func (att *Attestation) Container(
+	// Cache TTL for chainloop commands, in seconds
+	//  Defaults to 0: no caching
+	// +optional
+	// +default=0
+	ttl int,
+) *Container {
+	ctr := dag.
+		Container().
+		From(fmt.Sprintf("ghcr.io/chainloop-dev/chainloop/cli:%s", chainloopVersion)).
+		WithEntrypoint([]string{"/chainloop"}). // Be explicit to prepare for possible API change
+		WithEnvVariable("CHAINLOOP_DAGGER_CLIENT", chainloopVersion)
+	if att.Token != nil {
+		ctr = ctr.WithSecretVariable("CHAINLOOP_ROBOT_ACCOUNT", att.Token)
+	}
+	if att.Source != nil {
+		ctr = ctr.WithMountedDirectory(".", att.Source)
+	}
+	if addr := att.Registry.Address; addr != "" {
+		ctr = ctr.WithEnvVariable("CHAINLOOP_REGISTRY_SERVER", addr)
+	}
+	if user := att.Registry.Username; user != "" {
+		ctr = ctr.WithEnvVariable("CHAINLOOP_REGISTRY_USERNAME", user)
+	}
+	if pw := att.Registry.Password; pw != nil {
+		ctr = ctr.WithSecretVariable("CHAINLOOP_REGISTRY_USERNAME", pw)
+	}
+	// Cache TTL
+	ctr = ctr.WithEnvVariable("DAGGER_CACHE_KEY", time.Now().Truncate(time.Duration(ttl) * time.Second).String())
+	return ctr
+}
+
+// Generate, sign and push the attestation to the chainloop control plane
+func (att *Attestation) Push(ctx context.Context, key, passphrase *Secret) (string, error) {
+	return att.
+		Container(0).
+		WithMountedSecret("/tmp/key.pem", key).
+		WithSecretVariable("CHAINLOOP_SIGNING_PASSWORD", passphrase).
+		WithExec([]string{
+			"attestation", "push",
+			"--remote-state",
+			"--attestation-id", att.AttestationID,
+			"--key", "/tmp/key.pem",
+		}).Stdout(ctx)
+}
+
+// Mark the attestation as failed
+func (att *Attestation) MarkFailed(
+	ctx context.Context,
+	// The reason for canceling, in human-readable form
+	// +optional
+	reason string,
+) error {
+	return att.reset(ctx, "failure", reason)
+}
+
+// Mark the attestation as canceled
+func (att *Attestation) MarkCanceled(
+	ctx context.Context,
+	// The reason for canceling, in human-readable form
+	// +optional
+	reason string,
+) error {
+	return att.reset(ctx, "cancellation", reason)
+}
+
+// Call `chainloop reset` to mark the attestation as either failed or cancelled.
+func (att *Attestation) reset(ctx context.Context,
+	// +optional
+	// The trigger that caused the reset.
+	// May be "failure" or "cancellation"
+	trigger string,
+	// The reason for the reset, in human-readable form
+	// +optional
+	reason string,
+) error {
+	args := []string{
+		"attestation", "reset",
+		"--remote-state",
+		"--attestation-id", att.AttestationID,
+	}
+	if reason != "" {
+		args = append(args, "--reason", reason)
+	}
+	if trigger != "" {
+		args = append(args, "--trigger", trigger)
+	}
+	_, err := dag.
```

Comment:
I think there is a bug here 
```suggestion
	_, err := att.
```

Reply:
Thanks! fixed